### PR TITLE
add fluidkeys to user's crontab

### PR DESCRIPTION
**Pull request by paulfurley**
* add scheduler.Enable, Disable
* enable or disable cron (scheduler) from init
* add Config.RunFromCron()
* add Config.Filename()...
* add ui.go with printInfo ... this is a small step to factoring out some of the UI components

### CODE REVIEW

File: config/config.go [hunk 1]
```diff
@@ -57,6 +57,18 @@ func (c *Config) GetFilename() string {
 	return c.filename
 }
 
+func (c *Config) RunFromCron() bool {
```
Comment (idrysdale) [inline config/config.go]: Ah hah, cunning. So it looks like we're going to record in our `config` if we expect to be running from `cron` or not...

File: config/config.go [hunk 2]
```diff
@@ -213,14 +213,29 @@ type key struct {
...
+#     # store_password tells Fluidkeys to use the system keychain to store
+#     # the password for this key and look for it before prompting.
 #     store_password = true
+#
+#     # rotate_automatically specifies that key rotation tasks should be
+#     # carried out without prompting when running 'fk key rotate --automatic'
+#     # store_password must also be true to rotate keys automatically.
 #     rotate_automatically = true
+#
+# THIS FILE IS OVERWRITTEN BY FLUIDKEYS.
```
Comment (idrysdale) [inline config/config.go]: Ah, that's sensible. Good documentation 💪

File: scheduler/scheduler.go [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+	return removeCrontabLines(crontab) + cronLines
+}
+
+func removeCrontabLines(crontab string) string {
+	return strings.Replace(crontab, cronLines, "", -1)
+}
+
+const crontab string = "crontab"
+const cronLines string = `
+# Fluidkeys added the following line. To disable, edit ~/.config/fluidkeys/config.toml
```
Comment (idrysdale) [inline scheduler/scheduler.go]: IIRC Fluidkeys directory can be different. Is it possible to dynamically create this line for them to specify their actual directory?

Comment (paulfurley) [reply scheduler/scheduler.go]: Interesting point. The problem is that we want these two lines to stay *exaclty* the same between runs to make it trivial to parse them in an existing crontab. If they can change (including if we change them in the future) then we can't detect them anymore.

Maybe the thing to do is just say "To disable, edit your Fluidkeys configuration file"

Comment (idrysdale) [reply scheduler/scheduler.go]: 👍

**Pull request merged**